src/main/java/dev/openfeature/sdk/OpenFeatureClient.java

```diff
@@ -125,7 +126,13 @@ private <T> FlagEvaluationDetails<T> evaluateFlag(FlagValueType type, String key
                     defaultValue, provider, mergedCtx);
 
             details = FlagEvaluationDetails.from(providerEval, key);
-            hookSupport.afterHooks(type, hookCtx, details, mergedHooks, hints);
+            if (details.getErrorCode() != null) {
+                Exception e = ExceptionUtils.instantiateErrorByErrorCode(details.getErrorCode(),
+                        details.getErrorMessage());
+                hookSupport.errorHooks(type, hookCtx, e, mergedHooks, hints);
```

Comment:
The only sort-of weird edge case I can see here is if the error hooks themselves throw (unlikely) then they will possibly run again because of the catch block. Maybe we could have some boolean or some other way to track this?

It's a bit of an edge case, and maybe it could apply to the JS implementation too.

cc @Kavindu-Dodan @beeme1mr

Reply:
Good point, description of the error hook says,

> Run when evaluation encounters an error. This will always run. Errors thrown will be swallowed.

so we should avoid handling error from hook execution 🤔

Reply:
@toddbaert I added https://github.com/open-feature/java-sdk/pull/951/commits/31290014675e460bc42493ec80b9ae692d0cf30c to throw the derived exception so we have the same flow as if it comes from an exception. 

One could say this is an anti-pattern, but I think this is a valid usage

Reply:
Good solution, yep, a bit of an any pattern... if anything we could prevent the stack trace generation for performance, but that feels like a nice-to-have.